### ISSUE TITLE
Why hasn't the latest version of the XCFramework been signed?

### ISSUE THREAD

**Issue by XKCheng** [getsentry/sentry-cocoa#3740]
### Problem Statement

According to the guidelines for [verifying the origin of your XCFrameworks](https://developer.apple.com/documentation/Xcode/verifying-the-origin-of-your-xcframeworks), it is expected that XCFrameworks should be signed. However, in examining the [8.21.0](https://github.com/getsentry/sentry-cocoa/releases/tag/8.21.0) asset version, I could not locate any associated signature.

**Comment by brustolin** [comment]
Hello @XKCheng thanks for reaching out.

We have never signed any of our versions of the XCFramework. Our primary distribution methods are CocoaPods or SPM, and until now, none of these have had a pre-built framework.

The next version of SPM will have pre-built framework.
@philipphofmann I believe we should discuss about adopting framework signing.

**Comment by philipphofmann** [comment]
Related Apple docs: https://developer.apple.com/documentation/xcode/creating-a-multi-platform-binary-framework-bundle#Sign-the-XCFramework-bundle.

**Comment by iwan-ua** [comment]
I suspect this issue is the reason we can't update to Xcode 16
8.21.0 works fine on Xcode 15.4
On Xcode 16 it shows some errors, so I try to update it to 8.36.0
Build succeeds but installation of our system extension now fails with error:
`Domain=OSSystemExtensionErrorDomain Code=8 "Invalid code signature or missing entitlements"`

**Comment by brustolin** [comment]
Hello @iwan-ua, this error is only for the system extension? The main app is fine?

**Comment by iwan-ua** [comment]
> Hello [@iwan-ua](https://github.com/iwan-ua), this error is only for the system extension? The main app is fine?

Yes. The app runs when launched from Xcode. To run the system extension, you must first install it. We do it like this:
`let request = OSSystemExtensionRequest.activationRequest(forExtensionWithIdentifier: sysextBundleID, queue: DispatchQueue.global())`
 `OSSystemExtensionManager.shared.submitRequest(request)`
With updated sentry, this request instantly fails with 
`Domain=OSSystemExtensionErrorDomain Code=8 "Invalid code signature or missing entitlements"`

**Comment by iwan-ua** [comment]
I have a workaround now.
I deleted Sentry from SPM, and added manually downloaded latest Sentry.xcframework as "Do not embed". No problems with system extension now. I'm hoping to switch back to SPM when this issue is resolved.

**Comment by brustolin** [comment]
Thanks for the input @iwan-ua. We will look into finding a solution for the SPM

**Comment by antonshkurenko** [comment]
@brustolin thanks for your response, it's quite blocking thing once we upgrade to Sequoia and Xcode 16

**Comment by antonshkurenko** [comment]
Hi, any updates, guys?

**Comment by brustolin** [comment]
Sorry @antonshkurenko, no updates for this yet.

**Comment by armcknight** [comment]
@antonshkurenko sorry for delaying on this, but we do still plan to tackle it. I'm closing this issue because it's duplicated by #4391, please follow that one for updates.
